src/index.ts

```diff
@@ -0,0 +1,185 @@
+  installWrangler();
+  authenticationSetup();
+  await execCommands(getMultilineInput("preCommands"));
+  await uploadSecrets();
+  await genericCommand();
+  await execCommands(getMultilineInput("postCommands"));
+}
+
+function checkWorkingDirectory(workingDirectory = "") {
+  try {
+    const normalizedPath = path.normalize(workingDirectory);
+    if (existsSync(normalizedPath)) {
+      return normalizedPath;
+    } else {
+      setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+    }
+  } catch (error) {
+    setFailed(
+      `🚨 While checking/creating directory ${workingDirectory} received ${error}`
+    );
+  }
+}
+
+function installWrangler() {
+  startGroup("📥 Installing Wrangler");
+  const command = `npx install wrangler@${config["WRANGLER_VERSION"]}`;
+  info(`Running Command: ${command}`);
+  spawnSync(command, { cwd: config["workingDirectory"], env: process.env });
+  endGroup();
+}
+
+function authenticationSetup() {
+  startGroup("🔐 Authenticating with Cloudflare");
+  try {
+    process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_ACCOUNT_ID"];
+    process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_API_TOKEN"];
+    info(`Authentication process initiated with - API Token`);
+  } catch (error) {
+    setFailed(
+      `Authentication details were not found. Please input an 'apiToken' to the action.`
+    );
+  }
+  endGroup();
+}
+
+async function execCommands(commands: string[]) {
+  startGroup("🚀 Executing Pre/Post Commands");
+  if (!commands.length) {
+    warning(`📌 Pre/Post Commands were not provided, skipping execution.`);
+    return;
+  }
+  for (const command of commands) {
+    const npxCMD = command.startsWith("wrangler")
+      ? `npx ${command}`
+      : command;
+
+    info(`🚀 Executing command: ${npxCMD}`);
+
+    execSync(npxCMD, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+    });
+  }
+  endGroup();
+}
+
+async function uploadSecrets() {
+  startGroup("🔑 Uploading Secrets");
+  const secrets: string[] | string = config["secrets"];
+  if (!secrets.length) {
+    warning(`📌 No secrets were provided, skipping upload.`);
+    return;
+  }
+  const npxCMD =
+    process.env.RUNNER_OS === "Windows" ? "npx.cmd exec" : "npx";
+  const environment = config["ENVIRONMENT"];
+
+  const getSecret = (secret: string) => process.env[secret] ?? "";
+  const secretObj = secrets.reduce((acc: any, secret: string) => {
+    acc[secret] = getSecret(secret);
+    return acc;
+  }, {});
+
+  const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+  const secretCmd = `${npxCMD} wrangler secret:bulk ${JSON.stringify(
+    secretObj
+  )}${environmentSuffix}`;
+
+  try {
+    spawnSync(secretCmd, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+      stdio: "ignore"
+    });
+    info(`✅ Uploaded secrets`);
+  } catch (error) {
+    setFailed(`🚨 Failed to upload secrets`);
+  }
+
+  endGroup();
+}
+
+async function genericCommand() {
```

Comment:
Referring to this as "generic command" doesn't make sense to me. It's _very specifically_ a Wrangler command, right?

Reply:
What I mean is there's nothing particularly "generic" about it.